src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.Fixer.cs

```diff
@@ -0,0 +1,171 @@
+            SyntaxGenerator generator, ImmutableDictionary<string, string> properties, CancellationToken cancellationToken)
+        {
+            IArrayCreationOperation arrayArgument = GetArrayCreationOperation(node, model, cancellationToken, out bool isInvoked);
+            IEnumerable<ISymbol> typeFields = model.GetEnclosingSymbol(node.SpanStart, cancellationToken).ContainingType
+                .GetMembers().Where(x => x is IFieldSymbol);
+
+            // Get a valid member name for the extracted constant
+            string newMemberName = GetExtractedMemberName(typeFields.Select(x => x.Name), properties["paramName"]);
+
+            // Get method containing the symbol that is being diagnosed. Should always be in a method
+            IOperation? containingMethodBody = arrayArgument.GetAncestor<IMethodBodyOperation>(OperationKind.MethodBody);
+            if (containingMethodBody is null)
+            {
+                // This is due to VB methods having a different structure than CS methods
+                containingMethodBody = arrayArgument.GetAncestor<IBlockOperation>(OperationKind.Block);
+            }
+
+            // Create the new member
+            SyntaxNode newMember = generator.FieldDeclaration(
+                newMemberName,
+                generator.TypeExpression(arrayArgument.Type),
+                GetAccessibility(model.GetEnclosingSymbol(containingMethodBody!.Syntax.SpanStart, cancellationToken)),
+                DeclarationModifiers.Static | DeclarationModifiers.ReadOnly,
+                arrayArgument.Syntax
+            ).FormatForExtraction(containingMethodBody.Syntax);
+
+            // Add the new extracted member
+            ISymbol lastFieldSymbol = typeFields.LastOrDefault();
+            if (lastFieldSymbol is not null)
+            {
+                // Insert after last field if any fields are present
+                SyntaxNode lastFieldNode = await lastFieldSymbol.DeclaringSyntaxReferences.First().GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+                editor.InsertAfter(lastFieldNode, newMember);
+            }
+            else
+            {
+                // Insert before first method if no fields are present, as a method already exists
+                editor.InsertBefore(containingMethodBody.Syntax, newMember);
+            }
+
+            // Replace argument with a reference to our new member
+            SyntaxNode identifier = generator.IdentifierName(newMemberName);
+            if (isInvoked)
+            {
+                editor.ReplaceNode(node, generator.WithExpression(identifier, node));
+            }
+            else
+            {
+                editor.ReplaceNode(node, generator.Argument(identifier));
+            }
+
+            // Return changed document
+            return await Formatter.FormatAsync(editor.GetChangedDocument(), cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
Why do we want to format the whole document?

Reply:
I'll work on just formatting the node.